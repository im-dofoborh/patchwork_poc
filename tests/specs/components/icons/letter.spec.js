import { mount, createLocalVue } from '@vue/test-utils';
import LetterIcon from '~/components/icons/Letter';

describe('LetterIcon component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('letter-icon', LetterIcon);
    wrapper = mount(LetterIcon, {
      localVue,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders alternative size and colour variant', () => {
    wrapper.setProps({
      size: 100,
      colour: '#fff',
      textColour: '#000',
      word: 'Foo',
    })
    expect(wrapper.html()).toMatchSnapshot();
  });
});
