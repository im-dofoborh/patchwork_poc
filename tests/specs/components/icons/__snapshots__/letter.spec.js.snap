// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterIcon component renders alternative size and colour variant 1`] = `
<div title="Foo" class="sg-letter-icon" style="width: 100px;">
  <div class="sg-letter-icon__inner" style="height: 100px; width: 100px; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); font-size: 55.55555555555556px;">
    F
  </div>
</div>
`;

exports[`LetterIcon component renders default variant 1`] = `
<div title="Unknown" class="sg-letter-icon" style="width: 30px;">
  <div class="sg-letter-icon__inner" style="height: 30px; width: 30px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 16.666666666666668px;">
    U
  </div>
</div>
`;
